### PR TITLE
Fine cauldrons alignment.

### PR DESCRIPTION
**Pull request by dkorolev**
@mzhurovich : Great stuff with the optimizer, just a few minor things from my end after our chat.

Tested on `CPLUSPLUS` being `g++`/`clang++` and `NDEBUG` of `0/1`. All tests pass.

Thanks,
Dima

### CODE REVIEW

File: fncas/optimize.h [hunk 1]
```diff
@@ -145,14 +145,9 @@ OptimizationResult GradientDescentOptimizer<F>::Optimize(const std::vector<doubl
...
     const auto g = gi(current_point);
-    const auto try_step = [&dim, &current_point, &fi, &g ](double step)
-        -> std::pair<double, std::vector<double>> {
-      std::vector<double> candidate(dim);
-      for (size_t i = 0; i < dim; ++i) {
-        candidate[i] = current_point[i] - g.gradient[i] * step;
-      }
-      const double value = fi(candidate);
-      return std::make_pair(value, candidate);
+    const auto try_step = [&dim, &current_point, &fi, &g](double step) {
```
Comment (mzhurovich) [inline fncas/optimize.h:148]: Funny thing, but my clang-format changes `&g]` to `&g ]`

Comment (dkorolev) [inline fncas/optimize.h:148]: Hehe. Here's mine:

```
$ clang-format-3.5 --version
Ubuntu clang-format version 3.5.0-svn213690-1~exp1 (trunk) (based on LLVM 3.5.0)
```

**Comment by mzhurovich** [comment]
@dkorolev Cool! Thanks :)

**Comment by dkorolev** [comment]
Thank you!

Just realized you didn't have admin rights in `KnowSheet` until now. Should be fixed, do you have have permissions to merge now?

Thanks,
Dima

**Comment by dkorolev** [comment]
Pushed this pull request into `KnowSheet/master`. @mzhurovich : Please do `git fetch upstream ; git merge upstream/master` before making further changes.

(Somehow this pull request was from a `from_dkorolev` branch into a `from_dkorolev` branch. This is now fixed.)

Thanks again!
Dima

**Comment by dkorolev** [comment]
I should add that no immediate action is required. :+1:

**Pull request merged**